trackma/messenger.py

```diff
@@ -24,27 +26,40 @@
 class Messenger:
     _handler = None
 
-    def __init__(self, handler):
+    def __init__(self, handler, classname):
         self._handler = handler
+        self.classname = classname
 
     def set_handler(self, handler):
         self._handler = handler
 
-    def debug(self, classname, msg):
-        if self._handler:
-            self._handler(classname, TYPE_DEBUG, msg)
+    def with_classname(self, classname):
+        return Messenger(self._handler, classname)
 
-    def info(self, classname, msg):
+    def _call_handler(self, msgs, msg_type):
         if self._handler:
-            self._handler(classname, TYPE_INFO, msg)
+            cn, msg = self._parse_msgs(msgs)
+            self._handler(cn, msg_type, msg)
 
-    def warn(self, classname, msg):
-        if self._handler:
-            self._handler(classname, TYPE_WARN, msg)
+    def _parse_msgs(self, msgs):
+        if len(msgs) >= 2:
```

Comment:
Are there any instances left of the log functions being called with an explicit classname? Either way, to simplify the methods here, I'd rather design their signature to represent the 90% use case and add special invocation options for the other, such as a `name` kwarg. Especially with `exception` using a differnt signature already.

Reply:
> Are there any instances left of the log functions being called with an explicit classname?

Yes, the hooks. Including the example in [Hooks Doc](docs/hooks.rst). I thought that making these break would be quite bad, but i guess the crypt function signatures are even worse, i am fixing it.

Since these will be breaking changes, trackma version should bump up, probably?

Also, the `name` kwarg should be mandatory or it can be a positional argument?

```py
def debug(self, msg, name=None): pass # positional or as kword
# or 
def debug(self, msg, *, name=None): pass # only as kword
```